File: src/Jazor.CLR/module/IEqualityComparerT1Module.cs
namespace Jazor.CLR;

/// <summary>
/// System.Collections.Generic.IEqualityComparer&lt;T&gt; 模块映射规则
///
/// 目前仅开放与 EqualityComparer&lt;T&gt;.Default 可直接联动的 Equals 调用。
/// </summary>
[ECMAScriptModule("System/Collections/Generic/IEqualityComparerT1Module.js")]
[Jazor(Op.Alias, "System.Collections.Generic.IEqualityComparer<T>", "Object")]
public static class IEqualityComparerT1Module<T>
{
	/// <summary>
	/// C#: comparer.Equals(x, y)
	/// JS: 复用 EqualityComparer&lt;T&gt; 的核心等价逻辑
	/// 说明：保持实现简洁，不在此重复注入 instance 空检查。
	/// </summary>
	[Jazor(Op.Import, "System.Collections.Generic.IEqualityComparer<T>.Equals(T, T)")]
	public static bool _dae184550b995be1(object instance, T x, T y)
		=> EqualityComparerT1Module<T>.EqualsCore(x, y);
}

File: samples/RazorVue.TodoList/Todo.Host/wwwroot/jazor/System/Collections/Generic/IEqualityComparerT1Module.js
import { EqualityComparerT1Module } from "System/Collections/Generic/EqualityComparerT1Module.js";
export function _dae184550b995be1(instance, x, y) {
  EqualityComparerT1Module.ensureComparerInstance(instance);
  return EqualityComparerT1Module.equalsCore(x, y);
}
export function _f53ff8f6435182d7(instance, obj) {
  EqualityComparerT1Module.ensureComparerInstance(instance);
  return EqualityComparerT1Module.getHashCodeCore(obj);
}
export const IEqualityComparerT1Module = { _dae184550b995be1, _f53ff8f6435182d7 };
